agent/state.py

```
@@ -353,3 +381,61 @@ def update_result(self, session_id: str, plan: AgentPlan | dict, result: dict) -
+            if branch:
+                session.last_branch = branch
+            session.updated_at = _now()
+            with self._connect() as conn:
+                conn.execute(
+                    "UPDATE agent_sessions SET updated_at = ?, active_objective = ?, last_branch = ? WHERE session_id = ?",
+                    (session.updated_at, session.active_objective, session.last_branch, session_id),
+                )
+                conn.commit()
+            return AgentSession.model_validate(session.model_dump())
+
+    def update_resume_payload(self, session_id: str, payload: dict[str, Any] | None) -> AgentSession:
```

Ruff (0.15.13)

[error] Undefined name `Any`

(F821)